neo.UnitTests/Extensions/Nep5NativeContractExtensions.cs

```diff
@@ -13,41 +11,14 @@ namespace Neo.UnitTests.Extensions
 {
     public static class Nep5NativeContractExtensions
     {
-        internal class ManualWitness : IVerifiable
+        public static bool Transfer(this NativeContract contract, Persistence.Snapshot snapshot, byte[] from, byte[] to, BigInteger amount, bool signFrom)
         {
-            private readonly UInt160 _hashForVerify;
-
-            public Witness Witness
+            Transaction tx = new Transaction
             {
-                get => throw new NotImplementedException();
-                set => throw new NotImplementedException();
-            }
-
-            public int Size => 0;
-
-            public ManualWitness(UInt160 hashForVerify)
-            {
-                _hashForVerify = hashForVerify;
-            }
-
-            public void Deserialize(BinaryReader reader) { }
-
-            public void DeserializeUnsigned(BinaryReader reader) { }
+                Sender = signFrom ? from : new byte[20]
+            };
 
-            public UInt160 GetScriptHashForVerification(Persistence.Snapshot snapshot)
-            {
-                return _hashForVerify;
-            }
-
-            public void Serialize(BinaryWriter writer) { }
-
-            public void SerializeUnsigned(BinaryWriter writer) { }
-        }
-
-        public static bool Transfer(this NativeContract contract, Persistence.Snapshot snapshot, byte[] from, byte[] to, BigInteger amount, bool signFrom)
-        {
-            var engine = new ApplicationEngine(TriggerType.Application,
-                new ManualWitness(signFrom ? new UInt160(from) : null), snapshot, 0, true);
+            var engine = new ApplicationEngine(TriggerType.Application, tx, snapshot, 0, true);
```

Comment:
This is the moment we need to pass a limited execution mode for the transaction.